algebra/tests/matrix/tools.c

```diff
@@ -177,3 +177,31 @@ int algebraTests_diagCheck(matrix_t *M)
 
 	return CHECK_OK;
 }
+
+
+int algebraTests_trpCheck(matrix_t *M1, matrix_t *M2)
+{
+	int M1rows, M1cols;
+	int M2rows, M2cols;
+	int row, col;
+
+	M1rows = matrix_rowsGet(M1);
+	M1cols = matrix_colsGet(M1);
+
+	M2rows = matrix_rowsGet(M2);
+	M2cols = matrix_colsGet(M2);
+
+	if (M1rows != M2cols || M1cols != M2rows) {
+		return CHECK_FAIL;
+	}
```

Comment:
```suggestion
	unsigned int row, col, nRows, nCols;

	nRows = matrix_rowsGet(M1);
	nCols = matrix_colsGet(M1);

	if (nRows != matrix_colsGet(M2) || nCols != matrix_rowsGet(M2)) {
		return CHECK_FAIL;
	}
```

After the check you do not need `M2` colspan and rowspan. Not a blocker, but removing redundancies.